Taskfile.yaml

```diff
@@ -0,0 +1,93 @@
+    vars:
+      NESTED_MODULES: api
+      API_DIRS: '{{.ROOT_DIR}}/api/v1beta1/...'
+      MANIFEST_OUT: '{{.ROOT_DIR}}/api/crds/manifests'
+      CODE_DIRS: '{{.ROOT_DIR}}/cmd/... {{.ROOT_DIR}}/internal/... {{.ROOT_DIR}}/test/... {{.ROOT_DIR}}/api/v1beta1/...'
+      COMPONENTS: 'control-plane-operator'
+      REPO_URL: 'https://github.com/openmcp-project/control-plane-operator'
+      GENERATE_DOCS_INDEX: "true"
+      ENVTEST_REQUIRED: "true"
+      ENVTEST_K8S_VERSION: "1.30.0"
+  common: # imported a second time so that overwriting task definitions can call the overwritten task with a 'c:' prefix
+    taskfile: hack/common/Taskfile_controller.yaml
+    internal: true
+    aliases:
+    - c
+    vars:
+      NESTED_MODULES: api
+      API_DIRS: '{{.ROOT_DIR}}/api/v1beta1/...'
+      MANIFEST_OUT: '{{.ROOT_DIR}}/api/crds/manifests'
+      CODE_DIRS: '{{.ROOT_DIR}}/cmd/... {{.ROOT_DIR}}/internal/... {{.ROOT_DIR}}/test/... {{.ROOT_DIR}}/api/v1beta1/...'
+      COMPONENTS: 'control-plane-operator'
+      REPO_URL: 'https://github.com/openmcp-project/control-plane-operator'
+      GENERATE_DOCS_INDEX: "true"
+      ENVTEST_REQUIRED: "true"
+      ENVTEST_K8S_VERSION: "1.30.0"
+
+tasks:
+  test:
+    desc: "  Run all tests."
+    run: once
+    aliases:
+      - build:bin:val:test
+    vars:
+      NESTED_MODULES: api
+      API_DIRS: '{{.ROOT_DIR}}/api/v1beta1/...'
+      MANIFEST_OUT: '{{.ROOT_DIR}}/api/crds/manifests'
+      CODE_DIRS: '{{.ROOT_DIR}}/cmd/... {{.ROOT_DIR}}/internal/... {{.ROOT_DIR}}/test/... {{.ROOT_DIR}}/api/v1beta1/...'
+      COMPONENTS: 'control-plane-operator'
+      REPO_URL: 'https://github.com/openmcp-project/control-plane-operator'
+      GENERATE_DOCS_INDEX: "true"
+      ENVTEST_REQUIRED: "true"
+      ENVTEST_K8S_VERSION: "1.30.0"
+    deps:
+    - test-envtest-dep
+    cmds:
+    - 'PROJECT_ROOT="{{.ROOT_DIR2}}" NESTED_MODULES="{{.NESTED_MODULES}}" ENVTEST_K8S_VERSION="{{.ENVTEST_K8S_VERSION}}" {{.TASKFILE_DIR2}}/run-tests.sh {{.CODE_DIRS}}'
+
+  test-envtest-dep:
+    desc: "  Install the envtest dependency, if marked as required."
+    run: once
+    status:
+    - '[ "{{.ENVTEST_REQUIRED | default "false"}}" != "true" ]'
+    cmds:
+    - task: tools:envtest
+    vars:
+      ENVTEST_REQUIRED: "true"
+    internal: true
+
+  build-raw:
+    desc: "  Build the binary. Opposed to the regular build, this one just builds and skips code generation/validation tasks."
+    summary: |
+      This task builds the binary for the current operating system and architecture.
+      To overwrite this, set the 'OS' and 'ARCH' environment variables.
+      The binary is saved in the 'bin' folder, as $COMPONENT.$OS-$ARCH.
+    aliases:
+      - build:bin:build-raw
+    cmds:
+    - for:
+        var: COMPONENTS
+        as: COMPONENT
+      cmd: 'CGO_ENABLED=0 GOOS={{.OS}} GOARCH={{.ARCH}} go build -a -o {{.ROOT_DIR2}}/bin/{{.COMPONENT}}.{{.OS}}-{{.ARCH}} {{.ROOT_DIR2}}/cmd/main.go'
```

Comment:
```suggestion
```
I remember that I did this somewhere and you probably copied it from there, but I couldn't find it right now.
When I did this, the envtest stuff was not contained in the library, so it had to be added by these overwrites. By now, it is contained (see https://github.com/openmcp-project/build/blob/main/tasks_val.yaml#L23-L41), which means that simply setting `ENVTEST_REQUIRED` to `"true"` should be enough. Then, no overwrites are required, which means that the second include isn't required either (because it is only needed to refer to the original implementation in the overwrites).
This theory needs to be validated by running `task test` without the marked lines, though.

Reply:
The problem here is not envtest itself, the problem is that this particular instance expects the `ENVTEST_K8S_VERSION` environment variable which is not available to the `run-test.sh` script. running `task test` is what started all this :D